controller.py

```diff
@@ -469,16 +478,30 @@ def _make_request(url, on_404, on_timeout, on_exception, allow_401=False):
                 auth_url, problem
             )
             return problem
+
+        for rel, candidates in hyperlinks_to_create:
+            hyperlink, is_modified = library.set_hyperlink(rel, *candidates)
+            if is_modified:
+                # We need to send an email to this email address about
+                # what just happened. This is either so the receipient
+                # can confirm that the address works, or to inform
+                # them a new library is using their address.
+                hyperlink.notify(self.emailer, self.app.url_for)
```

Comment:
Setting the hyperlinks before calling library_catalog ensures that the hyperlinks show up as links in the catalog.